### PR TITLE
Make style prop types more flexible

### PR DESCRIPTION
**Pull request by hvergara**
*Description of changes:*
This PR makes style props more flexible: they always will accept an string, but without losing type inference & auto-completion ;)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/primitives/types/style.ts [hunk 1]
```diff
@@ -5,6 +5,13 @@ import { GridItemStyleProps, GridContainerStyleProps } from './grid';
 import { ImageStyleProps } from './image';
 import { TextAreaStyleProps } from './textArea';
 
+/**
+ * Extract string-like types. Defaulted to string if prop doesn't allow it
+ */
+export type StyleProp<PropertyType> = PropertyType extends string
```
Comment (hvergara) [inline packages/react/src/primitives/types/style.ts:11]: This is the important part: this generic type will extract the string-like types only, allowing to keep IDE auto-completion, and defaulting to string if no string-like type is found.

Examples:
- `StyleProp<"a" | "b" | "c"> => "a" | "b" | "c"`
- `StyleProp<"a" | "b" | "c" | number> => "a" | "b" | "c" | string`
- `StyleProp<"a" | "b" | "c" | string> => "a" | "b" | "c" | string`
- `StyleProp<any> => string`
- `StyleProp<unknown> => string`

Comment (reesscot) [reply packages/react/src/primitives/types/style.ts:11]: Can't we support `number` in addition to string?

Comment (hvergara) [reply packages/react/src/primitives/types/style.ts]: Fixed in last commit. Now number-compatible types are: `number | string | ("other" | "string" | "literals")`

**Review by dbanksdesign — APPROVED** [review]
LGTM! :shipit:

**Review by reesscot — APPROVED** [review]
LGTM!

**Review by reesscot — CHANGES_REQUESTED** [review]
Looks great, but needs a changeset before merging.

**Comment by hvergara** [comment]
> Looks great, but needs a changeset before merging.

Added :)

**Review by reesscot — APPROVED** [review]

**Pull request merged**